core/src/main/scala/org/apache/spark/SparkException.scala

```diff
@@ -128,13 +140,16 @@ private[spark] case class ExecutorDeadException(message: String)
 private[spark] class SparkUpgradeException(
     errorClass: String,
     errorSubClass: Option[String] = None,
-    messageParameters: Array[String],
+    messageParameters: Map[String, String],
```

Comment:
It is tested by at least QueryExecutionErrorsSuite "INCONSISTENT_BEHAVIOR_CROSS_VERSION: compatibility with Spark 2.4/3.2 in reading/writing dates".